File: 1048.c
#include <stdio.h>
int exist[505];
inline int min(int a,int b) { return a < b ? a : b; }
int main()
{
	int i,n,m,t,ans = 1000;
	scanf("%d%d",&n,&m);
	for (i = 0;i < n;++i) {
		scanf("%d",&t);
		if (t < m && exist[m - t] && (t < ans || m - t < ans))
			ans = min(t,m - t);
		exist[t] = 1;
	}
	if (ans < 1000)
		printf("%d %d\n",ans,m - ans);
	else
		puts("No Solution");
	return 0;
}

File: 1048.rb
# TLE
def min(a,b)
  a < b ? a : b
end
exist = []
ans = 1000
n, m = gets.chomp.split(' ').map &:to_i
a = gets.chomp.split(' ').reverse.map &:to_i
a.each do |x|
  ans = min(x,m - x) if x < m && m - x < 500 && exist[m - x] && (x < ans || m - x < ans)
  exist[x] = true
end
if ans < 1000
  puts "#{ans} #{m - ans}"
else
  puts 'No Solution'
end
